src/components/InfoWindow.js

```diff
@@ -0,0 +1,70 @@
+import React, { useContext } from 'react';
+import { getDistanceFromLatLonInKm } from '../helpers/Haversine';
+import { LocationsContext } from '../context/LocationsContext';
+import './InfoWindow.css';
+
+export default function InfoWindow({ $dimensionKey }) {
+  const locationsContext = useContext(LocationsContext);
+
+  const location = locationsContext.locations.filter(
```

Comment:
If you stick with the `filter` approach, consider renaming this variable `filteredLocations` and then assigning `const location = filteredLocations[0];` once instead of referencing `location[0]` throughout the rest of the function. I found it a little misleading to have `location` be an array when the fact it’s singular implies it should hold just 1 thing. Does that make sense?